src/java/org/apache/cassandra/db/partitions/PartitionIterators.java

```diff
@@ -83,6 +83,18 @@ public static PartitionIterator singletonIterator(RowIterator iterator)
         return new SingletonPartitionIterator(iterator);
     }
 
+    public static void consume(PartitionIterator iterator)
```

Comment:
do we need to close the given iterator?

Reply:
Good catch. This `consume` method was present in 3.0 but it was removed in 3.11 because there weren't usages of it, so we are reintroducing it. The original method didn't close the iterator because the callers did so with try-with-resources blocks. I'm preserving that behaviour and closing the iterator in the caller (`DataResolver#resolveWithReplicaFilteringProtection`).